apps/web-client/src/lib/db/orders/suppliers.ts

```diff
@@ -77,25 +76,28 @@ export async function getPossibleSupplerOrderInfos(db: DB): Promise<SupplierOrde
 	return result;
 }
 
-export async function createSupplierOrder(db: DB, orderLines: SupplierOrderLine[]): Promise<SupplierOrder[]> {
+/** @TODO Rewrite this function to accomodate for removing quantity in customerOrderLine */
+export async function createSupplierOrder(db: DB, orderLines: SupplierOrderLine[]) {
 	// Creates one or more supplier orders with the given order lines. Updates customer order lines to reflect the order.
 	// Returns one or more `SupplierOrder` as they would be returned by `getSupplierOrder`
+
 	const supplierOrderMapping = {};
 	// Collect all supplier ids involved in the order lines
 	const supplierIds = Array.from(new Set(orderLines.map((item) => item.supplier_id)));
+
 	await db.tx(async (passedDb) => {
 		const db: DB = passedDb as DB;
 		for (const supplierId of supplierIds) {
 			// Create a new supplier order for each supplier
-			const newId = (
+			const newSupplierOrderId = (
```

Comment:
Every time we have a query executed in a loop we should pause and examine if it's really really the only option we have. 99.99999% of the time (100% in my experience, but I consider myself open minded) it's better to add a new (or extend an existing) query to get all the needed data at once.

The idea behind SQL should be that a single query can get you what you need, efficiently.